services/main.go

```diff
@@ -648,3 +649,36 @@ func (server *ApiService) isOsVersion(params *omnitruck.RequestParams, c *fiber.
 	}
 	return fiber.NewError(fiber.StatusBadRequest, errMsg)
 }
+
+func (server *ApiService) downloadScriptHandler(c *fiber.Ctx) error {
+	params := getRequestParams(c)
+	c.Set("Content-Type", "application/x-sh")
+	err, ok := server.ValidateRequest(params, c)
+	if !ok {
+		server.logCtx(c).Error("Validation of download script API for " + params.Product + " failed")
+		return err
+	}
+	if params.OsType == "" {
+		return server.SendErrorResponse(c, http.StatusBadRequest, "query os_type can't be empty please use linux or windows depending on your operating system.")
+	}
+	if params.OsType != "linux" && params.OsType != "windows" {
+		return server.SendErrorResponse(c, http.StatusBadRequest, "query os_type can only be used as linux or windows depending on your operating system.")
```

Comment:
we should use `StatusUnprocessableEntity` as the request structure is not bad it's only the value which we don't support.

Reply:
Done